tui/network_dialog_check.go

```diff
@@ -0,0 +1,216 @@
+	dialog.progressBar.SetValue(dialog.progressMax)
+	clui.RefreshScreen()
+}
+
+// Failure is part of the progress.Client implementation and represents the
+// unsuccessful progress completion of a task by setting
+// the progress bar to "fail"
+func (dialog *NetworkTestDialog) Failure() {
+	bg := dialog.progressBar.BackColor()
+	flashTime := 100 * time.Millisecond
+	//dialog.progressBar.SetValue(0)  // leave the bar where it fails?
+	for i := 1; i <= 5; i++ {
+		dialog.progressBar.SetBackColor(term.ColorRed)
+		clui.RefreshScreen()
+		time.Sleep(flashTime)
+		dialog.progressBar.SetBackColor(bg)
+		clui.RefreshScreen()
+		time.Sleep(flashTime)
+	}
+}
+
+// Step is part of the progress.Client implementation and moves the progress bar one step
+// case it becomes full it starts again
+func (dialog *NetworkTestDialog) Step() {
+	if dialog.progressBar.Value() == dialog.progressMax {
+		dialog.progressBar.SetValue(0)
+	} else {
+		dialog.progressBar.Step()
+	}
+	clui.RefreshScreen()
+}
+
+// Desc is part of the progress.Client implementation and sets the progress bar label
+func (dialog *NetworkTestDialog) Desc(desc string) {
+	dialog.resultLabel.SetTitle(desc)
+	clui.RefreshScreen()
+}
+
+// Partial is part of the progress.Client implementation and adjusts the progress bar to the
+// current completion percentage
+func (dialog *NetworkTestDialog) Partial(total int, step int) {
+}
+
+// LoopWaitDuration is part of the progress.Client implementation and returns the time duration
+// each step should wait until calling Step again
+func (dialog *NetworkTestDialog) LoopWaitDuration() time.Duration {
+	return 1 * time.Second
+}
+
+// OnClose sets the callback that is called when the
+// dialog is closed
+func (dialog *NetworkTestDialog) OnClose(fn func()) {
+	clui.WindowManager().BeginUpdate()
+	defer clui.WindowManager().EndUpdate()
+	dialog.onClose = fn
+}
+
+// Close closes the dialog window and executes a callback if registered
+func (dialog *NetworkTestDialog) Close() {
+	clui.WindowManager().DestroyWindow(dialog.DialogBox)
+	clui.WindowManager().BeginUpdate()
+	closeFn := dialog.onClose
+	_ = term.Flush() // This might be dropped once clui is fixed
+	clui.WindowManager().EndUpdate()
+	if closeFn != nil {
+		closeFn()
+	}
+}
+
+func initDiaglogWindow(dialog *NetworkTestDialog) error {
+
+	const title = "Testing Networking..."
+	const wBuff = 5
+	const hBuff = 4
+	const dWidth = 50
+	const dHeight = 8
+
+	sw, sh := clui.ScreenSize()
+
+	x := (sw - TermWidth) / 2
+	y := (sh - TermHeight) / 2
+
+	posX := (TermWidth - dWidth + wBuff) / 2
+	if posX < wBuff {
+		posX = wBuff
+	}
+	posX = x + posX
+	posY := (TermHeight-dHeight+hBuff)/2 - hBuff
+	if posY < hBuff {
+		posY = hBuff
+	}
+	posY = y + posY
+
+	dialog.DialogBox = clui.AddWindow(posX, posY, dWidth, dHeight, title)
+	dialog.DialogBox.SetTitleButtons(0)
+	dialog.DialogBox.SetMovable(false)
+	dialog.DialogBox.SetSizable(false)
+	clui.WindowManager().BeginUpdate()
```

Comment:
again?

Reply:
This is how it is code in the PopUp dialog code in clui.